components/script/dom/dommatrixreadonly.rs

```diff
@@ -691,14 +690,18 @@ impl DOMMatrixReadOnlyMethods for DOMMatrixReadOnly {
     }
 
     // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-tofloat64array
-    #[allow(unsafe_code)]
-    fn ToFloat64Array(&self, cx: JSContext) -> NonNull<JSObject> {
-        let arr = self.matrix.borrow().to_array();
-        unsafe {
-            rooted!(in (*cx) let mut array = ptr::null_mut::<JSObject>());
-            let _ = Float64Array::create(*cx, CreateWith::Slice(&arr), array.handle_mut()).unwrap();
-            NonNull::new_unchecked(array.get())
-        }
+    fn ToFloat64Array(&self, cx: JSContext) -> Float64Array {
+        let vec: Vec<f64> = self
+            .matrix
+            .borrow()
+            .to_array()
+            .iter()
+            .map(|&x| x as f64)
+            .collect();
```

Comment:
Can we not just use `to_array`, since the matrix is typed to `f64`?

Reply:
Ah yes we can